File: day1/day1.py
datafile = open("input1.txt", "r")

sums = []
runningSum = 0
for line in datafile:
    if line == "\n":
        sums.append(runningSum)
        runningSum = 0
    else:
        runningSum += int(line)
sums.append(runningSum)
sums.sort(reverse=True)

print("Part 1 answer: ", sums[0])
print("Part 2 answer: ", sum(sums[0:3]))

File: day1/day1.hs
module Main where

import Data.List.Split ( splitOn )
import Data.List ( sort )

main :: IO ()
main = do
        nums <- readFile "day1/input1.txt"
        let food = (map sum . splitOn [0] . map numOrNull . lines) nums 
       
        putStr "Part 1 answer: " 
        print $ maximum food

        putStr "Part 2 answer: " 
        print $ (sum . take 3 . reverse . sort) food


numOrNull :: [Char] -> Int
numOrNull x 
    | null x = 0
    | otherwise = read x :: Int